xclim/indices/_agro.py

```diff
@@ -1079,17 +1081,214 @@ def _get_rain_season(pram):
+    """
+    # "WPM" method doesn't seem to work for gamma or pearson3
+    dist_and_methods = {"gamma": ["ML", "APP"], "fisk": ["ML", "APP"]}
+    if dist not in dist_and_methods:
+        raise NotImplementedError(f"The distribution `{dist}` is not supported.")
+    if method not in dist_and_methods[dist]:
+        raise NotImplementedError(
+            f"The method `{method}` is not supported for distribution `{dist}`."
+        )
+
+    if offset:
+        with xarray.set_options(keep_attrs=True):
+            da = da + convert_units_to(offset, da, context="hydro")
+
+    da, group = _preprocess_standardized_index(da, freq, window, **indexer)
+
+    def wrap_fit(da):
+        # if all values are null, obtain a template of the fit and broadcast
+        if indexer != {} and da.isnull().all():
+            select_dims = {d: 0 if d != "time" else [0, 1] for d in da.dims}
+            fitted = fit(da[select_dims], dist, method)
+            return fitted.broadcast_like(da.isel(time=0, drop=True))
+        return fit(da, dist, method)
+
+    params = da.groupby(group).map(wrap_fit)
+    cal_range = (
+        da.time.min().dt.strftime("%Y-%m-%d"),
+        da.time.max().dt.strftime("%Y-%m-%d"),
+    )
+    params.attrs = {
+        "Calibration period": cal_range,
+        "freq": freq,
+        "window": window,
+        "scipy_dist": dist,
+        "method": method,
+        "group": group,
+        "time_indexer": indexer,
+        "units": "",
+        "offset": offset,
+    }
+    return params
+
+
+def _get_standardized_index(da, params, **indexer):
+    """Compute standardized index for given fit parameters
+
+    This computes standardized indices which measure the deviation of  variables in the dataset compared
+    to a reference distribution. The reference is a statistical distribution computed with fitting parameters `params`
+    over a given calibration period of the dataset. Those fitting parameters are obtained with
+    ``xclim.standardized_index_fit_params``.
+
+    Parameters
+    ----------
+    da : xarray.DataArray
+        Input array.
+    params: xarray.DataArray
+        Fit parameters computed using ``xclim.indices.standardized_index_fit_params``.
+    indexer
+        Indexing parameters to compute the indicator on a temporal subset of the data.
+        It accepts the same arguments as :py:func:`xclim.indices.generic.select_time`.
+    """
+    scipy_dist = get_dist(params.attrs["scipy_dist"])
+    group_name = params.attrs["group"].rsplit(".")[-1]
+    param_names = params.dparams.values
+
+    # transform `da` values to a cdf distribution with fitted parameters `params`
+    # then invert back to a normal distrubtion (ppf)
+    def wrap_cdf_ppf(da, params, group_idxs):
+        if indexer != {} and np.isnan(da).all():
+            return np.zeros_like(da) * np.NaN
+
+        spi = np.zeros_like(da)
+        # loop over groups (similar to np.groupies idea, but np.groupies can't be used with custom functions)
```

Comment:
I had tried but got a confusing error message leading me to believe this claim was maybe false. But I will retry, it would be worth it to clean the function.